syntax = "proto3";

package phxrpc;

import "google/protobuf/descriptor.proto";
import "google/protobuf/wrappers.proto";


extend google.protobuf.MethodOptions {
    int32 CmdID = 2000000;
    string OptString = 2000001;
    string Usage = 2000002;
}

message MqttConnectPb {
    bool clean_session = 11;
    uint32 keep_alive = 12;
    string client_identifier = 13;
    string proto_name = 14;
    uint32 proto_level = 15;
}

message MqttConnackPb {
    bool session_present = 11;
    uint32 connect_return_code = 12;
}

message MqttPublishPb {
    bool dup = 1;
    uint32 qos = 2;
    bool retain = 3;
    uint32 packet_identifier = 11;
    string topic_name = 12;
    string content = 13;
}

message MqttPubackPb {
    uint32 packet_identifier = 11;
}

message MqttSubscribePb {
    uint32 packet_identifier = 11;
    repeated string topic_filters = 12;
    repeated uint32 qoss = 13;
}

message MqttSubackPb {
    uint32 packet_identifier = 11;
    repeated uint32 return_codes = 12;
}

message MqttUnsubscribePb {
    uint32 packet_identifier = 11;
    repeated string topic_filters = 12;
}

message MqttUnsubackPb {
    uint32 packet_identifier = 11;
}

message MqttPingreqPb {
}

message MqttPingrespPb {
}

message MqttDisconnectPb {
}

message HttpPublishPb {
    uint64 session_id = 1;
    MqttPublishPb mqtt_publish = 11;
}

message HttpPubackPb {
    uint64 session_id = 1;
    MqttPubackPb mqtt_puback = 11;
}

